Metalama.Framework.DesignTime/Utilities/DesignTimeExceptionHandler.cs

```diff
@@ -3,31 +3,45 @@
 using Metalama.Backstage.Diagnostics;
 using Metalama.Backstage.Extensibility;
 using Metalama.Backstage.Telemetry;
+using Metalama.Framework.Engine.Services;
 using Metalama.Framework.Engine.Utilities.Diagnostics;
+using Metalama.Framework.Services;
 
 namespace Metalama.Framework.DesignTime.Utilities
 {
-    public static class DesignTimeExceptionHandler
+    public class DesignTimeExceptionHandler : IGlobalService
     {
+        private readonly IExceptionReporter? _exceptionReporter;
+
+        public DesignTimeExceptionHandler(ServiceProvider<IGlobalService> serviceProvider)
+        {
+            this._exceptionReporter = serviceProvider.GetBackstageService<IExceptionReporter>();
+        }
+
         // It is critical that OperationCanceledException is NOT handled, i.e. this exception should flow to the caller, otherwise VS will be satisfied
         // with the incomplete results it received, and cache them. 
-        internal static bool MustHandle( Exception e )
+#pragma warning disable CA1822 // Mark members as static
+        internal bool MustHandle( Exception e )
```

Comment:
It semes it can be static.

Reply:
I didn't like making a service method static just because CA says so, but since it's just one class it's probably ok.